packages/vscode-extension/src/handlers.ts

```diff
@@ -468,21 +468,19 @@ export async function openFolder(
       commands.executeCommand("vscode.openFolder", folderPath);
     }, 2000);
   } else {
-    const autoOpenTimeout = setTimeout(() => {
-      commands.executeCommand("vscode.openFolder", folderPath, true);
-      ExtTelemetry.sendTelemetryEvent(TelemetryEvent.OpenNewProject, {
-        [TelemetryProperty.VscWindow]: VSCodeWindowChoice.NewWindowByDefault,
-      });
-    }, 5000);
-    const selection = await VS_CODE_UI.showMessage(
-      "info",
-      localize("teamstoolkit.handlers.openProject.title"),
-      false,
-      localize("teamstoolkit.handlers.openInNewWindow"),
-      localize("teamstoolkit.handlers.openInCurrentWindow")
-    );
+    const selection = await Promise.race([
+      new Promise<Result<string | undefined, FxError>>((resolve, reject) => {
+        setTimeout(resolve, 10000, err("timeout"));
+      }),
+      VS_CODE_UI.showMessage(
+        "info",
+        localize("teamstoolkit.handlers.openProject.title"),
+        false,
+        localize("teamstoolkit.handlers.openInNewWindow"),
+        localize("teamstoolkit.handlers.openInCurrentWindow")
+      ),
+    ]);
     if (selection.isOk()) {
-      clearTimeout(autoOpenTimeout);
```

Comment:
The 'vscode.openFoler' command will be executed in 2s. Will it time out to open in new window during this time if don't clear time out here?

Reply:
This PR uses `Promise.race()` instead of setTimeout&clearTimeout approach. It can be awaited.